tokio-tasks/src/lib.rs

```diff
@@ -1,6 +1,72 @@
+use std::future::Future;
+use std::pin::Pin;
+use std::task::Context;
+use std::task::Poll;
+use std::time::Duration;
+use tokio::time::error::Elapsed;
+use tokio::time::Timeout as TokioTimeout;
+use tracing::field;
+use tracing::Span;
+
+pub use future_ext::*;
 pub use task_map::*;
 pub use tasks::*;
 
 mod future_ext;
 mod task_map;
 mod tasks;
+
+#[tracing::instrument(name = "Sleep")]
+pub async fn sleep(duration: Duration) {
+    #[allow(clippy::disallowed_methods)]
+    tokio::time::sleep(duration).await
+}
+
+pub fn timeout<F>(duration: Duration, fut: F) -> Timeout<F>
+where
+    F: Future,
+{
+    #[allow(clippy::disallowed_methods)]
+    Timeout {
+        fut: tokio::time::timeout(duration, fut),
+        span: tracing::debug_span!(
+            "Future with timeout",
+            timeout_secs = duration.as_secs(),
+            timed_out = field::Empty
```

Comment:
this flag is really cool, I like it! we can quickly gather stats on timed out handlers 🦈